HIP/hip-0000-atomic-association-contracts.md

```diff
@@ -0,0 +1,115 @@
+---
+hip: 0000
+title: Optimstic token association for smart contracts
+author: Matthew DeLorenzo <mdelore.ufl@gmail.com>, Vae Vecturne <vea.vecturne@gmail.com>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+created: 2023-04-06
+updated: 2023-05-12
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/696
+---
+
+## Abstract
+
+Smart contracts are required to be associated with all tokens they custody. This makes sense when the contract has a long lived balance > 0; Hedera state is expanded and must be paid for in ongoing maintenance and storage costs to the network. However, there are many use cases that call for smart contracts atomically taking custody of a token, often to transfer to another entity within the same contract call. In this case, association is still required but Hedera state is not expanded.
```

Comment:
My perspective here is that if the contract ever held custody of an asset, it should pay the association cost out of sheer fairness to all entities in the network. Doing something else is giving special treatment to contracts when they are just another entity and should be treated as such.

Reply:
@littletarzan want to make sure you saw this comment ^^

Reply:
@mgarbs Is it true the association fee is to help pay for expanded network state size? If so, then atomic custody of tokens by smart contracts without association would not add to the net size of the network state. It seems unreasonable to have to pay the association fee to custody a token for an atomic or infinitesimal amount of time.